Sources/Testing/Events/Event.swift

```diff
@@ -143,37 +143,75 @@ extension Event {
   ///
   /// - Parameters:
   ///   - event: An event that needs to be handled.
-  public typealias Handler = @Sendable (Event) -> Void
+  ///   - context: The context associated with the event.
+  public typealias Handler = @Sendable (_ event: consuming Event, _ context: consuming Context) -> Void
+
+  /// A type which provides context about a posted ``Event``.
+  ///
+  /// An instance of this type is provided along with each ``Event`` that is
+  /// passed to an ``Event/Handler``.
+  public struct Context: Sendable {
+    /// The runner which ran the test which caused this instance's associated
+    /// event to be posted.
+    ///
+    /// This runner may be used within a ``Event/Handler`` to retrieve the
+    /// ``Test`` (if any) that a particular ``Event`` references via its
+    /// ``Event/testID`` property.
+    @_spi(ExperimentalTestRunning)
+    public var runner: Runner
+
+    /// Get the test which the specified event was associated with using this
+    /// context instance.
+    ///
+    /// - Parameters
+    ///   - event: The event whose associated test should be retrieved.
+    ///
+    /// - Returns: The ``Test`` which `event` was associated with, if any. If
+    ///   `event` was not associated with a particular test, `nil` is returned.
+    ///
+    /// This uses the event's ``Event/testID`` property to look up the
+    /// corresponding ``Test`` instance.
+    public func test(for event: Event) -> Test? {
+      guard let testID = event.testID else {
```

Comment:
This interface is awkward. Since the context for an event is specific to a particular test, we should just have that test as a stored property of the context rather than providing a lookup function.

We should also migrate the current test case in here for consistency.

Reply:
Yeah, I've taken this suggestion and now capture just the current `Test` and `Test.Case` values in this `Context` instance.